examples/confined_users-playbook.yml

```diff
@@ -0,0 +1,107 @@
+    selinux_logins:
+      # Map all new users to SElinux user "user_u" (instead of "unconfined_u")
+      - {login: '__default__', seuser: 'user_u', serange: 's0', state: 'present'}
+      # Alternatively map all new users to SElinux user "staff_u" (instead of "unconfined_u")
+      #- {login: '__default__', seuser: 'staff_u', serange: 's0-s0:c0.c1023', state: 'present'}
+      # Map admin users to SElinux user "sysadm_u"
+      - {login: 'root', seuser: 'sysadm_u', serange: 's0-s0:c0.c1023',
+               state: 'present'}
+      # Map select non-admin privileged users to SELinux user "staff_u"
+      #- {login: '<existing_user>', seuser: 'staff_u', serange: 's0', state: 'present'}
+
+  # Prepare the prerequisites required for this playbook
+  tasks:
+    # Enable kiosk mode -- add extremely limited users "guest" and "xguest"
+    #- name: Install xguest package
+    #  yum:
+    #    name: xguest
+    #    state: present
+
+    # Add new system users that will be assigned to staff_u SELinux user
+    - name: Add non-root privileged users
+      user:
+        comment: Linux System Roles SELinux User
+        name: "{{ item }}"
+        seuser: staff_u
+        group: "wheel"
+      loop: "{{ privileged_users }}"
+    # Allow ^^ to use sudo
+    - name: Allow privileged users to use sudo
+      copy:
+        dest: /etc/sudoers.d/{{ item }}
+        content: "{{ item }} ALL=(ALL) TYPE=sysadm_t ROLE=sysadm_r ALL"
+        mode: "0640"
+      loop: "{{ privileged_users }}"
+    # Run selinux role to apply changes described in "vars" section
+    - name: Execute the role and catch errors
+      block:
+        - name: Include selinux role
+          include_role:
+            name: linux-system-roles.selinux
+      rescue:
+        # Fail if failed for a different reason than selinux_reboot_required.
+        - name: Handle errors
+          fail:
+            msg: "role failed"
+          when: not selinux_reboot_required
+
+        - name: Restart managed host
+          reboot:
+
+        - name: Wait for managed host to come back
+          wait_for_connection:
+            delay: 10
+            timeout: 300
+
+        - name: Reapply the role
+          include_role:
+            name: linux-system-roles.selinux
+
+    # Users that got assigned a new SELinux user need to have their homedirs relabeled.
+    # Gather homedirs of each user with a new login mapping
+    - name: Gather homedirs of users with new login mapping
+      shell: >
+        getent passwd "{{ item.login }}" | awk -F: '{ print $6 }'
+      register: __login_homedirs
+      with_items: "{{ selinux_logins }}"
+    # Relabel the gathered homedirs (-F is necessary to change the user portion of file labels)
+    - name: Relabel homedirs of users with new login mapping
+      shell: "restorecon -RF {{ item.stdout }}"
+      when: item.stdout != ""
+      with_items: "{{ __login_homedirs.results if __login_homedirs.results is iterable else [] }}"
```

Comment:
Would it be possible to use https://github.com/linux-system-roles/selinux/blob/main/README.md#selinux_restore_dirs instead of `restorecon` command?

Reply:
This command needs to take place after the login mapping is applied. So we'd need to change order of tasks in tasks/main.yml.